applications/pelion_client/src/modules/pelion_trace.c

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright (c) 2021 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+ */
+
+#include "mbed-trace/mbed_trace.h"
+#include "mbed-trace-helper.h"
+
+#define MODULE pelion_trace
+#include <caf/events/module_state_event.h>
+
+
+static void init_pelion_trace(void)
+{
+	bool ret;
+#ifndef MCC_MINIMAL
+	// Create mutex for tracing to avoid broken lines in logs
+	ret = mbed_trace_helper_create_mutex();
+	__ASSERT_NO_MSG(ret);
+#endif
+	// Initialize mbed trace
+	(void) mbed_trace_init();
+	// If you want, you can filter out the trace set.
+	// For example  mbed_trace_include_filters_set("COAP");
+	// would enable only the COAP level traces.
+	// Or           mbed_trace_exclude_filters_set("COAP");
+	// would leave them out.
+	// More details on mbed_trace.h file.
+
+#ifndef MCC_MINIMAL
+	mbed_trace_mutex_wait_function_set(mbed_trace_helper_mutex_wait);
+	mbed_trace_mutex_release_function_set(mbed_trace_helper_mutex_release);
+#endif
+
+#if defined(__SXOS__) || defined(__RTX)
+	mbed_trace_print_function_set(trace_printer);
+#endif
+}
+
+
+static bool event_handler(const struct event_header *eh)
+{
+	if (is_module_state_event(eh)) {
+		struct module_state_event *event = cast_module_state_event(eh);
```

Comment:
We can use `const struct module_state_event *` since we do not modify event content.

Reply:
Updated